src/ObjectBuilder.cs

```diff
@@ -56,9 +56,11 @@ public virtual IModelType BuildServiceType(string serviceTypeName)
             type.XmlProperties = (SwaggerObject as Schema)?.Xml;
             type.Format = SwaggerObject.Format;
             var xMsEnum = SwaggerObject.Extensions.GetValue<JToken>(Core.Model.XmsExtensions.Enum.Name);
-            if ((SwaggerObject.Enum != null || xMsEnum != null) && type.KnownPrimaryType == KnownPrimaryType.String && !(IsSwaggerObjectConstant(SwaggerObject)))

```

Comment:
removing this check `type.KnownPrimaryType == KnownPrimaryType.String`. Since we want to support enums of different primary types.